### PR TITLE
Wire up redirects to execute against raft leader

### PR DESCRIPTION
**Pull request by pauldix**

### CODE REVIEW

File: services/meta/client.go [hunk 1]
```diff
@@ -399,34 +408,61 @@ func (c *Client) MarshalBinary() ([]byte, error) {
...
+			url = fmt.Sprintf("://%s/execute", server)
+			if c.tls {
+				url = "https" + url
+			} else {
+				url = "http" + url
+			}
+		}
+
+		index, err = c.exec(url, typ, desc, value)
+		tries++
```
Comment (dgnorton) [inline services/meta/client.go]: Shouldn't this get checked against `maxRetries`?

Comment (pauldix) [inline services/meta/client.go]: hah yes, forgot to add that part of it. Will do.

Comment (dgnorton) [inline services/meta/client.go]: also need to increment `currentServer`...and maybe `redirectServer` needs to be cleared in some cases?

Comment (pauldix) [inline services/meta/client.go]: redirect is cleared above right after it gets set. updating the `currentServer` increment

**Comment by corylanou** [comment]
+1

**Pull request merged**